src/main/java/org/dependencytrack/tasks/BaseTaskScheduler.java

```diff
@@ -0,0 +1,62 @@
+package org.dependencytrack.tasks;
+
+import alpine.common.logging.Logger;
+import alpine.event.framework.Event;
+import alpine.event.framework.EventService;
+import alpine.event.framework.SingleThreadedEventService;
+import com.asahaf.javacron.Schedule;
+
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+
+public abstract class BaseTaskScheduler {
+
+    private static final Logger LOGGER = Logger.getLogger(BaseTaskScheduler.class);
+
+    private Timer timer;
+
+    private static final long POLLING_INTERVAL_IN_MILLIS = 10000;
+
+    protected void scheduleTask(Map<Event, Schedule> eventScheduleMap) {
+        timer = new Timer();
+        timer.schedule(new ScheduleEvent().eventScheduleMap(eventScheduleMap), 100, 10000);
```

Comment:
Should we make the delay configurable, and use a higher default value than `100`?

Generally I think we should avoid scheduling any tasks while the API server is starting up, but the duration of the startup will of course vary depending on available resources etc.